processor/src/main/java/org/mapstruct/ap/internal/conversion/Conversions.java

```diff
@@ -222,6 +222,7 @@ private void registerJava8TimeConversions() {
         register( JavaTimeConstants.ZONED_DATE_TIME_FQN, Date.class, new JavaZonedDateTimeToDateConversion() );
         register( JavaTimeConstants.LOCAL_DATE_TIME_FQN, Date.class, new JavaLocalDateTimeToDateConversion() );
         register( JavaTimeConstants.LOCAL_DATE_FQN, Date.class, new JavaLocalDateToDateConversion() );
+        register( java.sql.Date.class, JavaTimeConstants.LOCAL_DATE_FQN, new JavaSqlDateToLocalDateConversion() );
```

Comment:
Why not do the reverser registration (first the `LOCAL_DATE_FQN` and then `java.sql.Date`)? Then you won't need the new register method. Also the name of the conversion would need to be `JavaLocalDateToSqlDateConversion`